Source/WebCore/platform/cocoa/MIMETypeRegistryCocoa.mm

```diff
@@ -92,14 +143,32 @@
 String MIMETypeRegistry::mimeTypeForExtension(StringView extension)
 {
     auto string = extension.createNSStringWithoutCopying();
-    return [[NSURLFileTypeMappings sharedMappings] MIMETypeForExtension:string.get()];
+
+    NSString *mimeType = [[NSURLFileTypeMappings sharedMappings] MIMETypeForExtension:string.get()];
+    if (mimeType.length)
+        return mimeType;
+
+    auto mapEntry = additionalMimeTypesMap().find<ASCIICaseInsensitiveStringViewHashTranslator>(extension);
+    if (mapEntry != additionalMimeTypesMap().end())
+        return mapEntry->value;
+
+    return nullString();
 }
 
 Vector<String> MIMETypeRegistry::extensionsForMIMEType(const String& type)
 {
     if (type.endsWith('*'))
         return extensionsForWildcardMIMEType(type);
-    return makeVector<String>([[NSURLFileTypeMappings sharedMappings] extensionsForMIMEType:type]);
+
+    NSArray *extensions = [[NSURLFileTypeMappings sharedMappings] extensionsForMIMEType:type];
+    if (extensions.count)
```

Comment:
Very optional nit suggestion to initialize the variable of interest in init statements for all the `if` expressions introduced in this PR, but ignore if you think it harms readability.

Reply:
Going to keep this as-is, unless you feel strongly, since I generally only use that pattern if the `if` expression is just a `nil`/`NULL` check. 

I think it is slightly more readable, avoiding a longer line of code, and makes it easier to add debug logging when needed.

Reply:
Sounds good, not a strong preference